pkg/agent/util/net_windows.go

```diff
@@ -897,60 +897,239 @@ func VirtualAdapterName(name string) string {
 	return fmt.Sprintf("%s (%s)", ContainerVNICPrefix, name)
 }
 
-func GetUplinkConfig(uplinkIfName string) (*config.AdapterNetConfig, error) {
-	iface, err := net.InterfaceByName(uplinkIfName)
+func MoveIFConfigurations(ips []*net.IPNet, routes []interface{}, mac net.HardwareAddr, mtu int, from string, to string) error {
+	return nil
+}
+
+func GetNetRoutesOnAdapter(linkIndex int) (string, []interface{}, error) {
+	var gw string
+	cmd := fmt.Sprintf("Get-NetRoute -InterfaceIndex %d -ErrorAction Ignore | Format-Table -HideTableHeaders", linkIndex)
+	rs, err := getNetRoutes(cmd)
 	if err != nil {
-		klog.ErrorS(err, "Failed to get interface", "uplinkIfName", uplinkIfName)
-		return nil, err
+		klog.ErrorS(err, "Failed to get routes")
```

Comment:
We generally want to avoid duplicated logs for a single error, and in many cases log in the caller (and when needed append the error message to the returned error, e.g. with fmt.Errorf()). You can check other code to learn more.

Reply:
Sure